### PR TITLE
Add chart download capability

### PR DESCRIPTION
**Pull request by martocodo**

### CODE REVIEW

File: src/Histogram.tsx [hunk 1]
```diff
@@ -105,6 +105,7 @@ export interface IHistogramProps {
   duration: number;
   colorScheme: string[];
   domain: IDomain;
+  downloadChartId?: string;
```
Comment (pollen8) [inline src/Histogram.tsx]: would this make sense to be changed just to 'id' ? Then we can reference the chart for other things than just downloading?

File: src/utils/outputSvg.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  height: number,
+  callback: (outpuData: string | Blob | null) => void,
+  type: TOutputType = 'blob',
+) => {
+  // Select the first svg element
+  const svg: any = select(`svg#${svgId}`);
+  const serializer = new XMLSerializer();
+  // generate IMG in new tab
+  const svgStr = serializer.serializeToString(svg.node());
+  const watermark = require('../assets/Powered-By-InfoSum_DARK.svg') as string;
```
Comment (pollen8) [inline src/utils/outputSvg.ts]: it would be nice to parametrize this option

File: src/utils/outputSvg.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        callback(pngData);
+      }
+    };
+    // start loading SVG data into in memory image
+    image.src = svgData;
+  }
+  // start loading watermark SVG data into memory
+  watermarkImage.src = watermark;
+};
+/* Alternate method for downloading image in new tab kept for reference
```
Comment (pollen8) [inline src/utils/outputSvg.ts]: do you want to keep these comments now?

Comment (martocodo) [reply src/utils/outputSvg.ts]: Not after you suggested the perfect download package to use instead :D

File: src/utils/outputSvg.ts (hunk 2 above)
Comment (martocodo) [reply src/utils/outputSvg.ts]: Good call - I'll add in params for the sizing of this too

File: src/Histogram.tsx (hunk 1 above)
Comment (martocodo) [reply src/Histogram.tsx]: Yep definitely - updating now

**Pull request merged**